### DISCUSSION TITLE
WPF app Works Xamarin doesn't

### DISCUSSION THREAD

**Discussion by MickCG** [Azure/azure-mobile-apps#512] [General]
OK, I am about to show my feet of clay here but I have the backend, data and WPF apps working. Or at least it seems to me that if the WPF app works for authentication, data access and sync, the Xamarin app should work if I copied and pasted the code correctly. But when the Android app or iPhone app runs I get unauthorized? Any idea fellow programmers where to look, especially have others recently got the Xamarin project working?

**Comment by adrianhall** [comment]
I have run through the xamarin forms tutorial within the last week.  

If you are using the SAME backend, you need to get to the authentication step; skip the "set up the web api for authentication" and continue on the client app setup (the application ID for the client in the AAD console and then the code to add the tokens).

**Reply by MickCG** [reply]
Thank you very much Adrian! Just before you replied I found where I was making a mistake in the     public class MsalActivity class I had left the brackets around my app ID, duh! I swear when something for me just doesn't work, it is always something simple!

**Reply by adrianhall** [reply]
I hear you.

I remember a time when I presented live coding on .NET Conf (a few years back) and was trying to copy some ID from my scratch pad to my code, and I kept copying the wrong one.  The entire chat was made up of people telling me "the NEXT one!" 

We've all been there.  I'm glad you got it sorted!
